src/general/PauseMenu.cs

```diff
@@ -69,28 +74,28 @@ public override void _Ready()
         saveMenu = GetNode<NewSaveMenu>(SaveMenuPath);
     }
 
-    public override void _UnhandledInput(InputEvent @event)
+    [RunOnKeyDown("ui_cancel")]
+    public void EscapeKeyPressed()
```

Comment:
I think the new input system should handle priority between this and transition manager. Godot inbuilt system with the other place calling SetInputAsHandled() properly only allows the top one of these two objects to succeed. Maybe RunOnKeyDown needs a new Exclusive attribute which stops event propagation when it is fired and the invoked method returns true (so this type of invoke would need to return a boolean, I suppose if the return type is void it can be assumed that the input is always consumed and unavailable to other input receivers).

Reply:
Do you mean something like https://github.com/Revolutionary-Games/Thrive/pull/1867/commits/91417cdf3c17442008d96fcb2f922177e715ff91 and https://github.com/Revolutionary-Games/Thrive/pull/1867/commits/d5ddfdf9202d88cf168d6b45e182ce37384bd952?

Reply:
From looking at those it seems you now have the part that the method can tell that it did or did not consume the key press. Did you already have the code for skipping any further callbacks once one consumes the input?